### PR TITLE
Support Wagtail 3

### PR DESCRIPTION
**Pull request by nickmoreton**
This changes the package so it is compatible with Wagtail v3.0

## Additions

Add new module import paths, including for Wagtail versions earlier than 3.0

## Removals

Remove compatibility with Wagtail versions earlier than 2.15

## Changes

Alter setup.py and tox to accommodate version chnages

## Testing

Fixed some tests

## Checklist

* [x] Changes are limited to a single goal (no scope creep)
* [x] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [x] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [x] Visually tested in supported browsers and devices
* [x] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### CODE REVIEW

**Comment by nickmoreton** [comment]
@spapas I've done it too 😄 But only because it was done against a forked repo so thought I'd add it here instead. It fixes the test etc.

My only thoughts are its Wagtail 2.15+ support so drops the older versions. If it's felt it needs to support a few versions back I can do that.

File: setup.py [hunk 1]
```diff
@@ -1,7 +1,8 @@
 from setuptools import find_packages, setup
 
 install_requires = [
-    "wagtail>=2.7,<3",
+    "wagtail>=2.15,<4.0",
+    "django>=3.0,<5.0",
```
Comment (willbarton) [inline setup.py:5]: Let's limit this to 3.2, the current LTS version, with appropriate changes elsewhere here.
```suggestion
    "django>=3.2,<5.0",
```

File: testmanage.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        execute_from_command_line(argv)
+    finally:
+        from wagtail.tests.settings import STATIC_ROOT, MEDIA_ROOT
+
+        shutil.rmtree(STATIC_ROOT, ignore_errors=True)
+        shutil.rmtree(MEDIA_ROOT, ignore_errors=True)
+
+
+if __name__ == "__main__":
+    runtests()
```
Comment (willbarton) [inline testmanage.py:66]: I don't understand the purpose of adding this file. Can you elaborate on its necessity?

If the intent is to add in deprecation warnings and fail on them, we already have a pattern we're using in another CFPB library, [Django-Flags](https://github.com/cfpb/django-flags/blob/main/tox.ini#L8) that I would prefer to see used, where we invoke `python` with:

```
python -W error::DeprecationWarning -W error::PendingDeprecationWarning
```

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -28,45 +28,36 @@ jobs:
...
+        include: 
+          - python: '3.9'
+            django: '3.2'
+            wagtail: '2.16'
+          - python: '3.10'
+            django: '3.2'
+            wagtail: '3.0'
+          - python: '3.10'
+            django: '4.0'
+            wagtail: '3.0'
```
Comment (willbarton) [inline .github/workflows/test.yml:45]: This set of changes (and to our tox setup below) give me pause. We try to keep our CFPB Python/Django/Wagtail libraries consistent with each other in their testing/CI setup to make maintenance that little bit easier for us (we're a small team, and we want to make the effort of maintaining these as separate libraries as simple as possible).

This is not a bad set of changes, but they're inconsistent with the other libraries we maintain. Changes like these should probably happen separately to a PR like this.

File: tox.ini [hunk 4]
```diff
@@ -2,29 +2,30 @@
 skipsdist=True
 envlist=
     lint,
-    py{36}-dj{22}-wag{27,latest},
-    py{39}-dj{22,32,40}-wag{latest}
+    python{3.7,3.8,3.9,3.10}-django{3.1,3.2,4.0}-wagtail{2.15,2.16,3.0}
```
Comment (willbarton) [inline tox.ini:5]: As with the GitHub Actions changes, while these might make sense in isolation, they're inconsistent with the other Python/Django/Wagtail libraries we maintain. I'd like to see wholesale changes to our test environments proposed in a separate PR.

**Review by willbarton — CHANGES_REQUESTED** [review]
The code changes here for Wagtail 3.0 compatibility look good to me.

I have some questions about the addition of the `testmanage.py` file below — I don't understand its necessity.

I also don't think I can approve this PR with the GitHub test workflow and tox changes you're proposing. As I describe further in my review, consistency across our CFPB libraries is important to us for maintaining them. That's not to say these are bad changes, it's just that I'd like to see them proposed in a way that's decoupled from adding Wagtail 3.0 support.

**Comment by willbarton** [comment]
On the consistency note, we've been stripping coveralls from our repos. If you don't mind, I'll open a quick PR to do that so that your contribution doesn't get stuck on this.

**Comment by nickmoreton** [comment]
> @nickmoreton Alright, coveralls has been removed in #56, so if you don't mind rebasing on main, that particular issue should go away.

Hi @willbarton thank you for your feedback. I should be able to look at this tomorrow 👍

File: testmanage.py (hunk 2 above)
Comment (nickmoreton) [reply testmanage.py]: Ah no reason other than I wanted to conveniently runs test locally without running tox. It's something I'm used to in our own packages. Happy to remove it 👍 and that was the plan.

File: setup.py (hunk 1 above)
Comment (nickmoreton) [reply setup.py]: I have made updates to this file to include some refinements.

File: tox.ini (hunk 4 above)
Comment (nickmoreton) [reply tox.ini]: I made changes here too along the lines of: https://github.com/cfpb/wagtail-flags/blob/main/tox.ini

**Comment by nickmoreton** [comment]
Hi @willbarton Thanks for you help here.

Can you review this again please. 
I think it's in a good place now my Github actions are running OK: https://github.com/nickmoreton/wagtail-sharing/actions/runs/2663659110

**Comment by willbarton** [comment]
@nickmoreton Yup, and I approved them to run here and they all pass!

**Review by willbarton — APPROVED** [review]
@nickmoreton thank you so much (and for your parallel work, @spapas!) for getting us Wagtail 3.0 compatibility, and thank you for being willing to match our existing conventions (though I really do like the GHA/tox env optimizations you had — it's worth us looking at adopting something like that).

This looks good to me! I'll get the required checks changed over and then get this merged, and we'll do a release!

**Pull request merged**